### DISCUSSION TITLE
Unable to connect to remote MSSQL server

### DISCUSSION THREAD

**Discussion by quinnjr** [prisma/prisma#22559] [Q&A]
### Question

I'm currently trying to get my client to connect to a remote MSSQL server but I keep erroring out with the Prisma client. My GUI Database tool, Beekeeper Studio, is able to make the connection successfully. As well, I've tried to connect to the server with telnet and netcat and both a successfully making the connection. Only Prisma seems to have the problem and I'm confused as to why.

### How to reproduce (optional)

1. Set DATABASE_URL in `.env` to `sqlserver://<host_ip>:1433;user=<user>;password=<password>;database=<database>;encrypt=true;trustServerCertificate=true;`
2. Attempt to create a client with `new PrismaClient()`
3. See error message:
```
PrismaClientInitializationError: Can't reach database server at `<host_ip>`:`1433`

Please make sure your database server is running at `<host_ip>`:`1433`.
```

### Expected behavior (optional)

I expect the client to successfully make the connection.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
import { Injectable, Logger, OnModuleInit } from '@nestjs/common';
import { PrismaClient } from '@prisma/client';

@Injectable()
export class DatabaseService extends PrismaClient implements OnModuleInit {
  private readonly logger = new Logger(DatabaseService.name);

  constructor() {
    super({
      log: (process.env['NODE_ENV'] == 'development') ? ['info'] : ['warn']
    });
  }

  public async onModuleInit() {
    this.$connect().catch(err => this.logger.error(err));
  }
}
```

- OS: ArchLinux
- Database: SQL Server
- Node.js version: v20.9.0

- Run `prisma -v` to see your Prisma version and paste it: v5.7.1

**Comment by quinnjr** [comment]
Adding logging to prisma gives me the following additional output:

```
prisma:info Starting a mssql pool with 49 connections.
prisma:info Performing a TLS handshake
prisma:info TLS handshake successful
PrismaClientInitializationError: Can't reach database server at `<database_ip>`:`1433`

Please make sure your database server is running at `<database_ip>`:`1433`.
```

**Comment by nurul3101** [comment]
Hello @quinnjr 👋 

Does your password contain any special characters?

Also, I found a related GitHub Issue, can you try specifying the instance name? 
- https://github.com/prisma/prisma/issues/6845

**Comment by quinnjr** [comment]
Password doesn't contain any special characters that need to be escaped.

We're able to make the connection with Knex.js without the instance name.

**Reply by ludralph** [reply]
Hi @quinnjr 

Can you double-check that there are no firewall rules blocking Prisma's connection attempts, even though other tools can connect.
You can also try increasing the connection timeout in your connection string:
```
sqlserver://<host_ip>:1433;user=<user>;password=<password>;database=<database>;encrypt=true;trustServerCertificate=true;connectionTimeout=30;
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
